internal/rpc/starknet/types_test.go

```diff
@@ -0,0 +1,276 @@
+			data: []byte(`"0"`),
+			want: BlockId{},
+			err:  ErrInvalidBlockId,
+		},
+		{
+			name: "zero number",
+			data: []byte(`0`),
+			want: BlockId{
+				value:  int64(0),
+				idType: blockIdNumber,
+			},
+			err: nil,
+		},
+		{
+			name: "positive number",
+			data: []byte(`1`),
+			want: BlockId{
+				value:  int64(1),
+				idType: blockIdNumber,
+			},
+			err: nil,
+		},
+		{
+			name: "pending tag",
+			data: []byte(`"pending"`),
+			want: BlockId{
+				idType: blockIdTag,
+				value:  "pending",
+			},
+			err: nil,
+		},
+		{
+			name: "latest tag",
+			data: []byte(`"latest"`),
+			want: BlockId{
+				idType: blockIdTag,
+				value:  "latest",
+			},
+			err: nil,
+		},
+		{
+			name: "block hash",
+			data: []byte(`"0x07d328a71faf48c5c3857e99f20a77b18522480956d1cd5bff1ff2df3c8b427b"`),
+			want: BlockId{
+				idType: blockIdHash,
+				value:  new(felt.Felt).SetHex("0x07d328a71faf48c5c3857e99f20a77b18522480956d1cd5bff1ff2df3c8b427b"),
+			},
+			err: nil,
+		},
+		{
+			name: "invalid block hash",
+			data: []byte(`"0x7d328a71faf48c5c3857e99f20a77b18522480956d1cd5bff1ff2df3c8b427b"`),
+			want: BlockId{},
+			err:  ErrInvalidBlockId,
+		},
+		{
+			name: "unexpected JSON type",
+			data: []byte(`{}`),
+			want: BlockId{},
+			err:  ErrInvalidBlockId,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			var got BlockId
+			if err := got.UnmarshalJSON(tt.data); !errors.Is(err, tt.err) {
+				t.Errorf("BlockId.UnmarshalJSON() error = %v, wantErr %v", err, tt.err)
+			}
+			assert.DeepEqual(t, got, tt.want, gocmp.Comparer(func(x, y BlockId) bool {
+				xv := reflect.ValueOf(x)
+				yv := reflect.ValueOf(y)
+				if xv.IsZero() || yv.IsZero() {
+					return xv.IsZero() && yv.IsZero()
+				}
+				if x.idType != y.idType {
+					return false
+				}
+				switch x.idType {
+				case blockIdHash:
+					return x.value.(*felt.Felt).Equal(y.value.(*felt.Felt))
+				case blockIdTag:
+					return x.value.(string) == y.value.(string)
+				case blockIdNumber:
+					return x.value.(int64) == y.value.(int64)
+				default:
+					return false
+				}
+			}))
+		})
+	}
+}
+
+func TestStorageKey_UnmarshalJSON(t *testing.T) {
+	tests := []struct {
+		name string
+		data []byte
+		want StorageKey
+		err  bool
+	}{
+		{
+			name: "empty data",
+			data: []byte{},
+			want: "",
+			err:  true,
+		},
+		{
+			name: "invalid storage key",
+			data: []byte(`"0x0ad328a71faf48c5c3857e99f20a77b18522480956d1cd5bff1ff2df3c8b427b"`),
+			want: "",
+			err:  true,
+		},
+		{
+			name: "valid storage key",
+			data: []byte(`"0x01d328a71faf48c5c3857e99f20a77b18522480956d1cd5bff1ff2df3c8b427b"`),
+			want: "0x01d328a71faf48c5c3857e99f20a77b18522480956d1cd5bff1ff2df3c8b427b",
+			err:  false,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			var got StorageKey
+			if err := got.UnmarshalJSON(tt.data); (err != nil) != tt.err {
+				t.Errorf("StorageKey.UnmarshalJSON(), unexpected error: %v", err)
```

Comment:
In keeping with the how the rest of the other failed tests are reported.

```suggestion
				t.Errorf("StorageKey.UnmarshalJSON() = %v, want nil", err)
```

Reply:
"want nil" isn't valid for all test cases, sometimes it is valid to return an error. Perhaps the best approach is:
```go
t.Errorf("StorageKey.UnmarshalJSON() = %v, unexpected error response", err)
```

Reply:
Ah. I see. We should also document that the key on line 155 is not valid because it's larger than `p`. That's the reason right?

Reply:
The reason is that doesn't match with the regex, but yes, isn't matching because is larger than `p`. I will add that doc to the test

Reply:
Done -> https://github.com/NethermindEth/juno/pull/377/commits/89b6a2a09b2d9609ac60b2904cf961e3fdd5ebd1